---
title: "BAT RNA-seq Analysis"
author: "Stefan Tholen, Kyle M. Kovary, Mary N. Teruel"
date: "Updated: `r format(Sys.time())`"
output: html_document
---

### Introduction

This is a notebook to document the analysis of the RNA-seq data collected by Stefan from brown adipose tissue in mice that were treated with corticosterone pellet or injections to combare to sham pellet or injections over the course of 14 days. This is still in development, eventually I plan have the data analysis pipeline with quality control and analysis plots as well as the code used to analyze everything.

```{r setup, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
library(tidyverse)
library(sleuth)
library(DESeq2)
library(readxl)
library(pheatmap)
library(RColorBrewer)
```

### Data Import

```{r data import, message=FALSE, warning=FALSE}
sample_list = read_xlsx('/Users/kylekovary/Box Sync/kkovary/R Projects/RNAseq/Stefan RNA-seq BAT analysis/Stefan Tholen_Samples_for_RNAseq_BAT.xlsx')

# Set up metadata table
path = list.dirs('/Users/kylekovary/Box Sync/kkovary/R Projects/RNAseq/Stefan RNA-seq BAT analysis/kallisto_output/')[-c(1,2)]
path = path %>% as.tibble() %>% 
  separate(col = value, into = c('a','mouse'), sep = 'ST-', remove = F) %>% 
  dplyr::select(value, mouse)

s2c = bind_cols(sample_list, path[match(sample_list$mouse, path$mouse),1]) %>%
  dplyr::rename(path = value, sample_name = `Sample Name`, Pellet = Treatment) %>% 
  unite('Treatment',c('Pellet','Delivery'), remove = F)

# Retrieve GeneNames
library(biomaRt)
tx2gene <- function(){
  mart <- biomaRt::useMart(biomart = "ensembl", dataset = "mmusculus_gene_ensembl")
  t2g <- biomaRt::getBM(attributes = c("ensembl_transcript_id", "ensembl_gene_id",
                                       "external_gene_name","entrezgene"), mart = mart)
  t2g <- dplyr::rename(t2g, target_id = ensembl_transcript_id,
                       ens_gene = ensembl_gene_id, ext_gene = external_gene_name)
  return(t2g)
}
t2g <- tx2gene()

so <- sleuth_prep(s2c, target_mapping = t2g, num_cores = max(1L, parallel::detectCores() - 1L), extra_bootstrap_summary = TRUE)

```

### DEseq2 Model

```{r message=FALSE, warning=FALSE}
full_design <- model.matrix(formula(~ s2c$Treatment + s2c$lane + s2c$Replicate, s2c$Day))

so <- sleuth_fit(so, full_design, 'full')
so <- sleuth_fit(so, formula = ~ s2c$lane + s2c$Replicate, fit_name = "reduced")
so <- sleuth_lrt(so, "reduced", "full")

plot_qq(so, test = 'reduced:full', test_type = 'lrt', sig_level = 0.05)

lrt_results <- sleuth_results(so, 'reduced:full', test_type = 'lrt')
table(lrt_results[,"qval"] < 0.05)
```

### Select Differentially Expressed Genes

```{r message=FALSE, warning=FALSE}
hits = filter(lrt_results, qval < 0.05)
hits_dat = so$obs_norm_filt %>% dplyr::select(target_id, sample, est_counts) %>% filter(target_id %in% hits$target_id) %>% spread(key = sample, value = est_counts)
genes = hits_dat$target_id
genes = hits[match(genes,hits$target_id),'ext_gene']
hits_dat = as.matrix(hits_dat[,2:ncol(hits_dat)])
rownames(hits_dat) = genes

zeros = which(hits_dat <= 0, arr.ind = T)
for(i in 1:nrow(zeros)){
  hits_dat[zeros[i,1], zeros[i,2]] = 1
}
```

### Heatmap of Differentially Expressed Genes

```{r message=FALSE, warning=FALSE}
hits_dat = t(apply(hits_dat, 1, function(x) x / mean(x)))

annotation = dplyr::select(s2c, Day, Treatment) %>% as.data.frame()
rownames(annotation) <- s2c$sample

mypal = rev(colorRampPalette(brewer.pal(11, "RdBu"))(n=200))

pheatmap(log2(hits_dat), breaks = seq(-3,3,6/(length(mypal))), 
         color = mypal, 
         show_rownames = F,
         show_colnames = F,
         annotation_col = annotation,
         cutree_cols = 3)

```


### PCA of Differentially Expressed Genes

```{r message=FALSE, warning=FALSE}
pca = prcomp(t(hits_dat), center = T, scale. = T)

pcaSum = summary(pca)$importance %>% t() %>% as.data.frame() %>% mutate(PC = rownames(.)) %>%
  gather(key = plot, value = value, 1:3)
pcaSum$PC <- factor(pcaSum$PC, levels = paste0('PC',1:ncol(hits_dat)))

ggplot(pcaSum, aes(x = PC, y = value)) + geom_point() +
  facet_wrap(~plot, scales = 'free', ncol = 1) + theme_classic() +
  theme(axis.text.x = element_text(angle = 90, hjust = 1)) + ylim(0, NA)


pcaDF = tibble(sample = as.numeric(rownames(pca$x)), PC1 = pca$x[,1], PC2 = pca$x[,2])
pcaDF = full_join(pcaDF, s2c, by = 'sample') %>% mutate(Day = as.factor(Day))

# 2D PCA between PC1 and PC2
ggplot(pcaDF, aes(x = PC1, y = PC2, colour = Pellet, size = Day, shape = Delivery)) + geom_point(alpha = 0.5) + theme_bw()

# 1D PCA plots since PC1 explains the majority of the variance
ggplot(gather(pcaDF,key = PC, value = value, PC1:PC2), aes(x = value, fill = Treatment)) + geom_dotplot(alpha = 0.5, binwidth = 5) + theme_bw() + facet_wrap(~PC, ncol = 1)
```













